server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -505,10 +511,6 @@ func (r *statsRecorder) handleTask(ctx context.Context, task *recordStatsTask) {
 func (r *statsRecorder) Stop() {
 	// Wait for all EventHandler channels to be closed to ensure there will be no
 	// more calls to Enqueue.
-	// TODO(bduffany): This has a race condition where the server can be shutdown
-	// just after the stream request is accepted by the server but before calling
-	// openChannels.Add(1). Can fix this by explicitly waiting for the gRPC server
-	// shutdown to finish, which ensures all streaming requests have terminated.
 	log.Info("StatsRecorder: waiting for EventChannels to be closed before shutting down")
 	r.openChannels.Wait()
```

Comment:
(sorry - just realized I misinterpreted the diff when I approved)

I think this `Wait()` call just needs to be replaced with the gRPCServerWaitGroup.Wait(), instead of doing it in `BuildEventHandler.Stop()` ? the `Wait()` here is where the race is happening

Reply:
Thanks (and done)! Nice to get rid of `openChannels`.